standards/application/webrtc.md

```diff
@@ -0,0 +1,331 @@
+
+**Privacy considerations:** Directly using a unique identifier (like a full peer ID) in the content topic can potentially allow observers to correlate messages with a specific user, since Waku relay or filter nodes will see the content topic of messages. To mitigate this, implementations **MAY** employ strategies such as hashing the peer ID or using bucketed topics (e.g. using a prefix of a hash of the peer ID) so that multiple peers share the same content topic, increasing anonymity. Another approach is to simply use a single, shared content topic for all WebRTC signaling (as was done in the Waku Phone prototype), embedding addressing info inside the message payload. A unified topic maximizes anonymity (all peers’ signals look alike) at the cost of efficiency (all clients receive all offers and must filter them). The choice of content topic strategy should balance privacy, scalability, and simplicity for the given application. In any case, encryption (discussed below) is critical so that only the intended recipient can read the signaling content, regardless of topic structure.
+
+For the purposes of this specification, we assume a content topic prefix of:
+
+```text
+/waku-webrtc/1/signal/proto
+```
+
+and, if using peer-specific topics, appending or including an identifier for the target peer. All messages **MUST** be published with the Protocol Buffers (proto) encoding for the payload, as defined in the Wire Protocol section. Peers **SHOULD** mark these signaling messages as ephemeral (non-persisted) when sending, so that Waku nodes do not store them long-term (they are transient by nature, similar to volatile signaling in calls).
+
+## Wire Protocol
+
+Signaling messages are encoded in a single Protocol Buffers message type. All messages share this format, distinguished by a message type field. The protobuf schema (proto3) is defined below:
+
+```proto
+syntax = "proto3";
+
+package waku.webrtc.v1;
+
+message WakuWebRTCMessage {
+  enum MessageType {
+    OFFER = 0;       // Initiator to responder: proposes a new WebRTC connection
+    ANSWER = 1;      // Responder to initiator: accepts and answers an OFFER
+    CANDIDATE = 2;   // Either peer to the other: conveys an ICE candidate (Trickle ICE, RFC 8838)
+    REJECT = 3;      // Responder to initiator: rejects an OFFER (connection refused)
+    BUSY = 4;        // Responder to initiator: rejects an OFFER because the responder is busy
+    END = 5;         // Either peer to the other: terminates an ongoing session (hang-up or cancel)
+    // (Future extensions could add ACK or other message types if needed)
+  }
+
+  MessageType message_type = 1;
+  bytes session_id = 2;               // Unique identifier for this session (generated by initiator)
+  optional bytes initiator_peer_id = 10;  // Waku peer ID of the initiator (caller)
+  optional bytes responder_peer_id = 11;  // Waku peer ID of the responder (callee)
```

Comment:
Does it have to be peer ids, can't it be an arbitrary ids agreed between applications out of band?